iidm/iidm-criteria/src/main/java/com/powsybl/iidm/criteria/duration/IntervalTemporaryDurationCriterion.java

```diff
@@ -0,0 +1,143 @@
+        private boolean highClosed = false;
+
+        /**
+         * Define the lower bound of the interval.
+         * @param value value of the lower bound (in seconds).
+         * @param closed <code>true</code> if the bound is part of the interval, <code>false</code> otherwise.
+         * @return the current builder
+         */
+        public Builder setLowBound(int value, boolean closed) {
+            checkValue(value);
+            checkBounds(value, highBound);
+            this.lowBound = value;
+            this.lowClosed = closed;
+            return this;
+        }
+
+        /**
+         * Define the upper bound of the interval.
+         * @param value value of the upper bound (in seconds).
+         * @param closed <code>true</code> if the bound is part of the interval, <code>false</code> otherwise.
+         * @return the current builder
+         */
+        public Builder setHighBound(int value, boolean closed) {
+            checkValue(value);
+            checkBounds(lowBound, value);
+            this.highBound = value;
+            this.highClosed = closed;
+            return this;
+        }
+
+        private void checkValue(int value) {
+            if (value <= 0) {
+                throw new IllegalArgumentException("Invalid bound value (must be > 0).");
+            }
+        }
+
+        private void checkBounds(Integer low, Integer high) {
+            if (low != null && high != null && low > high) {
+                throw new IllegalArgumentException("Invalid interval bounds values (low must be <= high).");
+            }
+        }
+
+        public IntervalTemporaryDurationCriterion build() {
+            if (lowBound == null && highBound == null) {
+                throw new IllegalArgumentException("Invalid interval criterion: at least one bound must be defined.");
+            }
+            return new IntervalTemporaryDurationCriterion(lowBound, highBound, lowClosed, highClosed);
+        }
+    }
+
+    /**
+     * Return a builder to create an {@link IntervalTemporaryDurationCriterion}.
+     * @return a builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public TemporaryDurationCriterionType getComparisonType() {
+        return TemporaryDurationCriterionType.INTERVAL;
+    }
+
+    @Override
+    public boolean filter(int acceptableDuration) {
+        boolean lowBoundOk = lowBound == null || acceptableDuration > lowBound
+                || lowClosed && acceptableDuration == lowBound;
+        boolean highBoundOk = highBound == null || acceptableDuration < highBound
+                || highClosed && acceptableDuration == highBound;
+        return lowBoundOk && highBoundOk;
+    }
+
+    /**
+     * Get the lower bound (in seconds) of the interval.
+     * @return lower bound of the acceptable interval (in seconds), or <code>null</code> if the interval has no lower bound.
+     */
+    public Integer getLowBound() {
```

Comment:
Another thing : it can be null should it be a Optional ?

Reply:
same for all the bounds

Reply:
Done